pkg/volume/rbd/rbd_util.go

```diff
@@ -724,17 +724,26 @@ func (util *RBDUtil) rbdInfo(b *rbdMounter) (int, error) {
 		return 0, fmt.Errorf("can not get image size info %s: %s", b.Image, string(output))
 	}
 
-	return getRbdImageSize(output)
+	outputs := strings.Split(string(output), "\n")
+
+	return getRbdImageSize(outputs)
 }
 
-func getRbdImageSize(output []byte) (int, error) {
+func getRbdImageSize(outputs []string) (int, error) {
+	var err error
 	info := struct {
 		Size int64 `json:"size"`
 	}{}
-	if err := json.Unmarshal(output, &info); err != nil {
-		return 0, fmt.Errorf("parse rbd info output failed: %s, %v", string(output), err)
+
+	for _, output := range outputs {
+		if err = json.Unmarshal([]byte(output), &info); err != nil {
```

Comment:
a better fix is to only use the stdout of `b.exec.Command` in line 707
rbd info prints informational data on stderr, but with --format=json as used the json lands on stdout which then can easily be parsed

Reply:
I agree - it looks error prone to me to just consume lines with parseable json and throw away the other ones without notice. What if rbd info emits multiline json?